llama_stack/providers/tests/datasetio/test_datasetio.py

```diff
@@ -95,7 +95,7 @@ async def test_register_dataset(self, datasetio_stack):
         assert len(response) == 1
         assert response[0].identifier == "test_dataset"
 
-        with pytest.raises(Exception) as exc_info:
+        with pytest.raises(Exception):  # noqa: B017
```

Comment:
ooh interesting -- can you actually replace this by a better Exception class instead of adding a noqa? It seems like `pytest.raises(Exception)` is just evil because a typo in the code will still make this part pass.

Reply:
Before this I think the warning was ignored so I'm not really changing the behavior, just avoiding the unused variable.

I can dig a little deeper like I did for the other parts of the code, I believe ValueError might be sufficient. Thanks!

Reply:
@leseb yeah I think a ValueError should be sufficient here agreed